lib/tts_isolate.dart

```diff
@@ -24,14 +27,50 @@ Future<void> isolateMain(
   await Firebase.initializeApp();
   final ttsQueue = TTSQueue();
 
-  // listen for changes to the tts preferences
-  bool isPreludeMuted = false;
+  final ttsModel = TtsModel.fromJson(
+      jsonDecode(prefs.getString("tts", defaultValue: '{}').getValue()));
+
+  // listen for changes to the tts preferences and update the isolates ttsModel
   final ttsPrefs = prefs.getString('tts', defaultValue: '{}');
-  ttsPrefs.listen((value) {
+  ttsPrefs.listen((value) async {
     final Map<String, dynamic> ttsMap = jsonDecode(value);
```

Comment:
Does something like

```
final ttsModel = TtsModel.fromJson(jsonDecode(value));
```

work here? That would allow avoiding adding all the setters.

Also I realized this suffers from the same shared preferences caching issue you mentioned before, you may have to route the preferences through the method channel instead of listening on shared preferences.